src/pages/docs/step-ca/configuration.mdx

```diff
@@ -384,6 +386,32 @@ for supported databases:
   </div>
 </Alert>
 
+### PostgreSQL
```

Comment:
Is a server TLS connection to PostgreSQL supported? And if so, can we describe how to write the DSN for it, and where to get the root CA certificate?

My [PostgreSQL PZT](https://smallstep.com/practical-zero-trust/postgresql-tls) might be of some benefit here, or [this part of the libpq-ssl docs](https://www.postgresql.org/docs/current/libpq-ssl.html#LIBQ-SSL-CERTIFICATES) (do the same params and defaults apply to Go's psql driver?)

Reply:
Good call. TLS is supported. One example of a DSN is this:

`postgresql://user:password@127.0.0.1:5432/mydb?sslmode=verify-full`

Its behavior is similar to what's in `libpq`. By default it will use the system root CAs as retrieved by Go's `x509.SystemCertPool()` to verify connections to the server. The library will also try default settings for the root CA file and client key / cert pair. It'll look for a directory called `(.)postgresql` in the user home directory to find `root.crt`, `postgresql.key` and `postgresql.crt` (as described here too: https://www.postgresql.org/docs/current/libpq-ssl.html#LIBQ-SSL-CERTIFICATES). If these files do not exist, they'll not be used in the configuration. If they are found, the'll be used to configure trusted roots and the client certificate.

It's also possible to explicitly provide a root CA file and/or a private key / cert pair:

`postgresql://user:password@127.0.0.1:5432/mydb?sslmode=verify-full&sslrootcert=/path/to/ca/file.pem&sslcert=/path/to/client.pem&sslkey=/path/to/client.key`

This DSN would result in providing a client certificate at connection setup time, the certificates in the root CA file being loaded as trusted roots and performing server hostname verification (incl. certificate chain verification using the configured trusted roots).

Haven't tried these configurations yet, but I read the code for parsing the DSN, especially the TLS part, at https://github.com/jackc/pgconn/blob/ded272b1f2c31e345b3209d7ab373822bf90b761/config.go#L559-L673.

In general the library mimics the behavior of `libpq` and also supports most of the same parameters. The library also reads parameters from environment variables (e.g. `PGHOST`, `PGUSER`, `PGSSLMODE`, etc; again similar to `libpq`), which are merged with defaults and the DSN. The DSN takes precedence over the environment variables.

What parts of this would you like me to add to the docs? I could also refer to your blog and/or the Postgres docs for more background, of course.

Reply:
At minimum, we should let folks know:
- that a TLS connection is possible, and how to enable basic server cert verification with `sslmode=verify-full`
- that it will trust the system CA store
- that it mimics `libpq` behavior, with a link to their [docs for TLS support](https://www.postgresql.org/docs/current/libpq-ssl.html) for more info.
- and as a bonus, you could show the more complex example of a DSN that explicitly provides a root ca, client cert and client key, and they can pick. I don't expect those parameters will change often, so even though they are for postgres and not step-ca, I think it's OK to include them. (Our docs also have several PKCS#11 URIs for various HSM platforms, for example)

Reply:
I've added a new `TLS` subheader to describe these settings.

Reply:
![giphy-1](https://user-images.githubusercontent.com/50809/159763256-385d89ab-71ff-4f9b-802e-8d9906a5bc25.gif)